### PR TITLE
Minor: improve documentation to StringView trim

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Follow on to https://github.com/apache/datafusion/pull/12395 from @Rachelint 

## Rationale for this change

Part of reviewing https://github.com/apache/datafusion/pull/12395I spent some non trivial time reading this code, and I wanted to encodde my learnings as comments

## What changes are included in this PR?

1. Add comments to various functions
2. Update parameter names

## Are these changes tested?
No code, just docs and name changes

## Are there any user-facing changes?
No, this is all internal code

### CODE REVIEW

File: datafusion/functions/src/string/common.rs [hunk 1]
```diff
@@ -35,19 +35,26 @@ use datafusion_expr::ColumnarValue;
 
 /// Append a new view to the views buffer with the given substr
 ///
-/// raw must be a valid view
-/// substr must be a valid substring of raw
+/// # Safety
+/// original_view must be a valid view
```
Comment (comphead) [inline datafusion/functions/src/string/common.rs:39]: what is valid view? 🤔

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb

File: datafusion/functions/src/string/common.rs (hunk 1 above)
Comment (alamb) [reply datafusion/functions/src/string/common.rs]: Good call -- I tried to clarify in c8668df3b

**Comment by alamb** [comment]
Thanks again @comphead

**Pull request merged**